lms/services/canvas.py

```diff
@@ -0,0 +1,42 @@
+from lms.services.exceptions import CanvasFileNotFoundInCourse
+
+
+class CanvasService:
+    """A high level Canvas service."""
+
+    api = None
+
+    def __init__(self, canvas_api):
+        self.api = canvas_api
+
+    def public_url_for_file(self, file_id, course_id, check_in_course=False):
+        """
+        Get a public URL for a Canvas file.
+
+        This will also attempt to check if the file is in the course to detect
+        course copy situations and fix it if we can.
+
+        :param file_id: The file to look up
+        :param course_id: The course the file should be in
+        :param check_in_course: Enable pre-emptive checking for the file in the
+            course regardless of whether the user has permission to see it
+        :return: A URL suitable for public presentation of the file
+        :raises  CanvasFileNotFoundInCourse: If we determine the file should
+            be in the course but isn't.
+        """
+
+        if check_in_course:
+            if not self._file_in_course(file_id, course_id):
+                # Looks like a course copy
+                raise CanvasFileNotFoundInCourse(file_id)
+
+        return self.api.public_url(file_id)
```

Comment:
Not much here yet, but for course copy this is going to get big.

Also honestly, the `file_in_course` function probably was never appropriate to be on the Canvas API object in the first place. I think this is potentially a result of a misapplication of our rules:

 * The views must be thin
 * Therefore this function can't go in there
 * Therefore it should go in the Canvas API

I'd say a few counter points there are:

 * Our views don't need to be so thin they can't have a list comprehension in them
 * Even if the functionality can't go in the view it doesn't imply it should stick to the closest service either